src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.ConnectionPooling.cs

```diff
@@ -0,0 +1,99 @@
+                using (var serverReader = new StreamReader(serverStream))
+                using (var serverWriter = new StreamWriter(serverStream))
+                {
+                    await serverReader.ReadLineAsync(); // GET line
+                    await serverReader.ReadLineAsync(); // blank line
+                    await serverWriter.WriteAsync(responseBody);
+                    await serverWriter.FlushAsync();
+                    await firstRequest;
+
+                    Task<Socket> secondAccept = listener.AcceptAsync();
+
+                    Task<string> additionalRequest = client.GetStringAsync(uri);
+                    await Task.Yield(); // TODO: This yield should not be necessary, but it is with WinHttpHandler for some reason.
+
+                    await serverReader.ReadLineAsync(); // GET line
+                    await serverReader.ReadLineAsync(); // blank line
+                    await serverWriter.WriteAsync(responseBody);
+                    await serverWriter.FlushAsync();
+                    await additionalRequest;
+
+                    Assert.False(secondAccept.IsCompleted, $"Second accept should never complete");
+                }
+            }
+        }
+
+        [Fact]
+        public async Task ServerDisconnectsAfterInitialRequest_SubsequentRequestUsesDifferentConnection()
+        {
+            using (var client = new HttpClient())
+            using (var listener = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp))
+            {
+                listener.Bind(new IPEndPoint(IPAddress.Loopback, 0));
+                listener.Listen(100);
+                var ep = (IPEndPoint)listener.LocalEndPoint;
+                var uri = new Uri($"http://{ep.Address}:{ep.Port}/");
+
+                string responseBody =
+                    "HTTP/1.1 200 OK\r\n" +
+                    $"Date: {DateTimeOffset.UtcNow:R}\r\n" +
+                    "Content-Length: 0\r\n" +
+                    "\r\n";
+
+                // Make multiple requests iteratively.
+                for (int i = 0; i < 3; i++)
+                {
+                    Task<string> request = client.GetStringAsync(uri);
+                    using (Socket server = await listener.AcceptAsync())
+                    using (var serverStream = new NetworkStream(server, ownsSocket: false))
+                    using (var serverReader = new StreamReader(serverStream))
+                    using (var serverWriter = new StreamWriter(serverStream))
+                    {
+                        await serverReader.ReadLineAsync(); // GET line
+                        await serverReader.ReadLineAsync(); // blank line
+
+                        await server.SendAsync(new ArraySegment<byte>(Encoding.ASCII.GetBytes(responseBody)), SocketFlags.None);
+
+                        await request;
+                        server.Shutdown(SocketShutdown.Both);
```

Comment:
It seems like there's a potential timing issue here.  The "initial read" may not necessarily complete immediately, and so we won't detect that the socket is closed until we try to write to it.  

This is actually a general issue with our connection pooling scheme that we're not addressing right now.